File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.js
var findThePrefixCommonArray = function (A, B) {
    const n = A.length
    const result = new Array(n)
    for (let i = 0; i < n; i++) {
        let count = 0
        for (let A_i = 0; A_i <= i; A_i++) {
            for (let B_i = 0; B_i <= i; B_i++) {
                if (A[A_i] === B[B_i]) {
                    count++
                    break
                }
            }
        }
        result[i] = count
    }
    return result
};

File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size();
        vector<int> result(n);
        vector<bool> isPresentInA(n + 1, false); // isPresent[i] = true, means i is present in A
        vector<bool> isPresentInB(n + 1, false);

        for (int i = 0; i < n; i++) {
            isPresentInA[A[i]] = true;
            isPresentInB[B[i]] = true;

            int count = 0;
            for (int num = 1; num <= n; num++) {
                if (isPresentInA[num] == true && isPresentInB[num] == true) {
                    count++;
                }
            }
            result[i] = count;
        }
        return result;
    }
};